app/models/user.rb

```diff
@@ -7,25 +7,15 @@ def name
     [first_name, last_name].join(' ')
   end
 
-  class << self
-    #
-    # For GDPR caseworker personal data is not stored in the event stream.
-    # Rendering an application's history can require many user names to be
-    # found. This method caches those lookups.
-    #
-    # It returns "[deleted]" for a forgotten/not found user
-    #
-    def name_for(id)
-      Rails.cache.fetch("user_names#{id}", expires_in: 10.minutes) do
-        User.find(id).name
-      rescue ActiveRecord::RecordNotFound
-        I18n.t('values.deleted_user_name')
-      end
-    end
+  def deactivated?
+    deactivated_at.present?
```

Comment:
Is there a chance we might reactivate a user? and this value remains?

Reply:
Good point. I'll change this to be a boolean and capture any activation deactivation timestamps in a log.